> /lgtm cancel for the UT failure which needs to be fixed.

@cici37 this test seems need to be updated for the existing behavior 

https://github.com/kubernetes/kubernetes/blob/5d28e86245509725523863143a5d9288418976ad/pkg/controller/namespace/deletion/namespaced_resources_deleter_test.go#L405-L467

```

k8s.io/kubernetes/pkg/controller/namespace: deletion expand_less | 0s
-- | --
{Failed  === RUN   TestDeleteEncounters404     namespaced_resources_deleter_test.go:449: ns1 {{ns1 delete-collection /v1, Resource=pods } { } {{ } <nil> nil <nil> <nil> [] <nil>} {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:449: ns1 {{ns1 list /v1, Resource=pods } fake-metadata-client-group/v1, Kind=  { } {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:449: ns1 {{ns1 delete-collection example.com/v1, Resource=flakes } { } {{ } <nil> nil <nil> <nil> [] <nil>} {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:449: ns1 {{ns1 list example.com/v1, Resource=flakes } fake-metadata-client-group/v1, Kind=  { } {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:449: ns1 {{ns1 list example.com/v1, Resource=flakes } fake-metadata-client-group/v1, Kind=  { } {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:451: ns1: expected delete-collection -> fallback to list -> list to verify 0 items     namespaced_resources_deleter_test.go:463: ns2 {{ns2 delete-collection /v1, Resource=pods } { } {{ } <nil> nil <nil> <nil> [] <nil>} {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:463: ns2 {{ns2 list /v1, Resource=pods } fake-metadata-client-group/v1, Kind=  { } {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:463: ns2 {{ns2 delete-collection example.com/v1, Resource=flakes } { } {{ } <nil> nil <nil> <nil> [] <nil>} {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:463: ns2 {{ns2 list example.com/v1, Resource=flakes } fake-metadata-client-group/v1, Kind=  { } {{ }   false false   <nil> 0  <nil>}}     namespaced_resources_deleter_test.go:465: ns2: expected delete-collection -> list to verify 0 items --- FAIL: TestDeleteEncounters404 (0.01s)


```

--- a/pkg/controller/namespace/deletion/namespaced_resources_deleter.go
+++ b/pkg/controller/namespace/deletion/namespaced_resources_deleter.go
@@ -528,12 +528,10 @@ func (d *namespacedResourcesDeleter) deleteAllContent(ctx context.Context, ns *v
 		gvrToNumRemaining:        map[schema.GroupVersionResource]int{},
 		finalizersToNumRemaining: map[string]int{},
 	}
+	podsGVR := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.OrderedNamespaceDeletion) {
-		// TODO: remove this log when the feature gate is enabled by default
-		logger.V(5).Info("Namespace controller - OrderedNamespaceDeletion feature gate is enabled", "namespace", namespace)
+	if _, hasPods := groupVersionResources[podsGVR]; hasPods && utilfeature.DefaultFeatureGate.Enabled(features.OrderedNamespaceDeletion) {
 		// Ensure all pods in the namespace are deleted first
-		podsGVR := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
 		gvrDeletionMetadata, err := d.deleteAllContentForGroupVersionResource(ctx, podsGVR, namespace, namespaceDeletedAt)
 		if err != nil {
 			errs = append(errs, fmt.Errorf("failed to delete pods for namespace: %s, err: %w", namespace, err))
@@ -561,6 +559,10 @@ func (d *namespacedResourcesDeleter) deleteAllContent(ctx context.Context, ns *v
 
 	// Proceed with deleting other resources in the namespace
 	for gvr := range groupVersionResources {
+		if utilfeature.DefaultFeatureGate.Enabled(features.OrderedNamespaceDeletion) && gvr.Group == podsGVR.Group &&
+			gvr.Version == podsGVR.Version && gvr.Resource == podsGVR.Resource {
+			continue
+		}
 		gvrDeletionMetadata, err := d.deleteAllContentForGroupVersionResource(ctx, gvr, namespace, namespaceDeletedAt)
 		if err != nil {
 			// If there is an error, hold on to it but proceed with all the remaining

--- a/pkg/features/versioned_kube_features.go
+++ b/pkg/features/versioned_kube_features.go
@@ -545,7 +545,8 @@ var defaultVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate
 	},
 
 	OrderedNamespaceDeletion: {
-		{Version: version.MustParse("1.33"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.30"), Default: false, PreRelease: featuregate.Beta},
+		{Version: version.MustParse("1.33"), Default: true, PreRelease: featuregate.Beta},
 	},
 
 	PersistentVolumeLastPhaseTransitionTime: {

--- a/test/featuregates_linter/test_data/versioned_feature_list.yaml
+++ b/test/featuregates_linter/test_data/versioned_feature_list.yaml
@@ -864,7 +864,11 @@
   versionedSpecs:
   - default: false
     lockToDefault: false
-    preRelease: Alpha
+    preRelease: Beta
+    version: "1.30"
+  - default: true
+    lockToDefault: false
+    preRelease: Beta
     version: "1.33"
 - name: PersistentVolumeLastPhaseTransitionTime
   versionedSpecs:

